### DISCUSSION TITLE
1.8.5版本mongodb没开启认证情况下执行工单报错

### DISCUSSION THREAD

**Discussion by jimsoft1000** [hhyo/Archery#1893] [Ideas]
1.8.5版本mongodb没开启认证情况下执行工单报错，发现是源代码里，应该是强行认证了用户名和密码，建议做如下修改
多一步判断，没有用户名和密码的时候，调用mongo命令的时候，就不输入用户名和密码的参数，这样执行，就不会报错了

        if self.port and self.host:
            msg = ""
            auth_db = self.instance.db_name or 'admin'
            try:
                if not sql.startswith('var host='): #在master节点执行的情况
                    if self.user and self.password:
                        cmd = "{mongo} --quiet -u {uname} -p '{password}' {host}:{port}/{auth_db} <<\\EOF\ndb=db.getSiblingDB(\"{db_name}\");{slave_ok}printjson({sql})\nEOF".format(mongo=mongo, uname=self.user, password=self.password, host=self.host, port=self.port, db_name=db_name, sql=sql, auth_db=auth_db, slave_ok=slave_ok)
                    else:
                        cmd = "{mongo} --quiet {host}:{port}/{auth_db} <<\\EOF\ndb=db.getSiblingDB(\"{db_name}\");{slave_ok}printjson({sql})\nEOF".format(mongo=mongo, host=self.host, port=self.port, db_name=db_name, sql=sql, auth_db=auth_db, slave_ok=slave_ok)
                else:
                    if self.user and self.password:
                        cmd = "{mongo} --quiet -u {user} -p '{password}' {host}:{port}/{auth_db} <<\\EOF\nrs.slaveOk();{sql}\nEOF".format(mongo=mongo, user=self.user, password=self.password, host=self.host, port=self.port, db_name=db_name, sql=sql, auth_db=auth_db)
                    else:
                        cmd = "{mongo} --quiet {host}:{port}/{auth_db} <<\\EOF\nrs.slaveOk();{sql}\nEOF".format(mongo=mongo, host=self.host, port=self.port, db_name=db_name, sql=sql, auth_db=auth_db)
                logger.debug(cmd)

**Comment by LeoQuote** [comment]
直接提pr吧
